anndata/_core/views.py

```diff
@@ -126,6 +126,7 @@ def as_view(obj, view_args):
 
 
 @as_view.register(np.ndarray)
+@as_view.register(DaskArray)
```

Comment:
I think this might need it's own view class. 

* It would be nice to know that it's a view of a dask array at a type level.
* Views are being loaded into memory:

```python
import anndata as ad
import dask.array as da
import numpy as np


a = ad.AnnData(
    np.ones((5, 10), dtype=np.float32),
    obsm={"dask": da.random.random((5, 20))}
)
display(a.obsm["dask"])
```

<img width="425" alt="image" src="https://user-images.githubusercontent.com/8238804/190443065-a38ed63d-cf1e-4b6f-9f86-19718d38cd31.png">

```python
view = a[:2]
display(view.obsm["dask"])
```

```
ArrayView([[0.51141181, 0.80505702, 0.72229628, 0.8530678 , 0.84690909,
            0.1601411 , 0.97500309, 0.4163151 , 0.79551743, 0.2428621 ,
            0.50578874, 0.47671809, 0.02821872, 0.99338238, 0.81772692,
            0.84456787, 0.13378153, 0.58699808, 0.17240616, 0.87100689],
           [0.98989405, 0.17072794, 0.94920147, 0.93408043, 0.23794826,
            0.82444377, 0.0869244 , 0.55049418, 0.94464469, 0.79332537,
            0.67213581, 0.41819875, 0.0634449 , 0.0353851 , 0.78107282,
            0.94189   , 0.710782  , 0.35208005, 0.90897056, 0.67785694]])
```

```python
b = view.copy()
display(b.obsm["dask"])
```

```
array([[0.51141181, 0.80505702, 0.72229628, 0.8530678 , 0.84690909,
        0.1601411 , 0.97500309, 0.4163151 , 0.79551743, 0.2428621 ,
        0.50578874, 0.47671809, 0.02821872, 0.99338238, 0.81772692,
        0.84456787, 0.13378153, 0.58699808, 0.17240616, 0.87100689],
       [0.98989405, 0.17072794, 0.94920147, 0.93408043, 0.23794826,
        0.82444377, 0.0869244 , 0.55049418, 0.94464469, 0.79332537,
        0.67213581, 0.41819875, 0.0634449 , 0.0353851 , 0.78107282,
        0.94189   , 0.710782  , 0.35208005, 0.90897056, 0.67785694]])
```

Reply:
Ok, in the recent commit the behavior is like this:
First output looks the same.

```python
view = a[:2]
display(view.obsm["dask"])
```
```
DaskArrayView([[0.67451575, 0.6255451 , 0.29382805, 0.32638486,
                0.94882371, 0.96557258, 0.32915445, 0.68920697,
                0.82376327, 0.32344818, 0.65802747, 0.90954457,
                0.89987801, 0.44695522, 0.68305018, 0.97012088,
                0.45402617, 0.8436169 , 0.30581718, 0.66890336],
               [0.00803059, 0.12620636, 0.48831987, 0.91531449,
                0.23068138, 0.33108069, 0.88164289, 0.19952604,
                0.35021352, 0.90056961, 0.49632834, 0.72886952,
                0.69350378, 0.98114544, 0.35685473, 0.52749217,
                0.55908432, 0.5112443 , 0.07758866, 0.92408913]])
```
```python
b = view.copy()
display(b.obsm["dask"])
```
```
array([[0.67451575, 0.6255451 , 0.29382805, 0.32638486, 0.94882371,
        0.96557258, 0.32915445, 0.68920697, 0.82376327, 0.32344818,
        0.65802747, 0.90954457, 0.89987801, 0.44695522, 0.68305018,
        0.97012088, 0.45402617, 0.8436169 , 0.30581718, 0.66890336],
       [0.00803059, 0.12620636, 0.48831987, 0.91531449, 0.23068138,
        0.33108069, 0.88164289, 0.19952604, 0.35021352, 0.90056961,
        0.49632834, 0.72886952, 0.69350378, 0.98114544, 0.35685473,
        0.52749217, 0.55908432, 0.5112443 , 0.07758866, 0.92408913]])
```
Did I get this right or should the outputs look different somehow